deltachat-ios/Chat/AppPickerViewController.swift

```diff
@@ -0,0 +1,153 @@
+import UIKit
+import WebKit
+
+protocol AppPickerViewControllerDelegate: AnyObject {
+    func pickedAnDownloadedApp(_ viewController: AppPickerViewController, fileURL: URL)
+}
+
+class AppPickerViewController: UIViewController {
+    // Web view
+    // Context
+    weak var delegate: AppPickerViewControllerDelegate?
+    let webView: WKWebView
+    var defaultCloseButton: UIBarButtonItem?
+    let downloadingView: DownloadingView
+
+    init(url: URL = URL(string: "https://webxdc.org/apps/")!) {
+        webView = WKWebView(frame: .zero)
+        webView.translatesAutoresizingMaskIntoConstraints = false
+        webView.load(URLRequest(url: url))
+
+        downloadingView = DownloadingView()
+        downloadingView.translatesAutoresizingMaskIntoConstraints = false
+        downloadingView.isHidden = true
+
+        super.init(nibName: nil, bundle: nil)
+
+        webView.navigationDelegate = self
+        view.addSubview(webView)
+        view.addSubview(downloadingView)
+        view.backgroundColor = .systemBackground
+        setupConstraints()
+        let closeButton = UIBarButtonItem(image: UIImage(systemName: "xmark"), style: .plain, target: self, action: #selector(AppPickerViewController.close(_:)))
```

Comment:
as discussed, for consistency, better use `UIBarButtonItem(barButtonSystemItem: .cancel ...)`